openedx/core/djangoapps/schedules/management/commands/tests/test_send_upgrade_reminder.py

```diff
@@ -88,3 +85,39 @@ def test_filter_out_verified_schedules(self):
                 message.context['course_ids'],
                 [str(schedules[i].enrollment.course.id) for i in (1, 2, 4)]
             )
+
+    @ddt.data(
+        ExperienceTest(experience=ScheduleExperience.DEFAULT, offset=expected_offsets[0], email_sent=True),
+        ExperienceTest(experience=ScheduleExperience.COURSE_UPDATES, offset=expected_offsets[0], email_sent=False),
+    )
+    @patch.object(tasks, 'ace')
+    def test_upgrade_reminder_experience(self, test_config, mock_ace):
+        current_day, offset, target_day, upgrade_deadline = self._get_dates(offset=test_config.offset)
+
+        schedule = self._schedule_factory(
+            offset=offset,
+            experience__experience_type=test_config.experience,
+        )
+
+        self.task.apply(kwargs=dict(
+            site_id=self.site_config.site.id, target_day_str=serialize(target_day), day_offset=test_config.offset,
+            bin_num=self._calculate_bin_for_user(schedule.enrollment.user),
+        ))
+
+        self.assertEqual(mock_ace.send.called, test_config.email_sent)
+
+    @patch.object(tasks, 'ace')
+    def test_upgrade_reminder_without_experience(self, mock_ace):
+        current_day, offset, target_day, upgrade_deadline = self._get_dates(offset=self.expected_offsets[0])
+
+        schedule = self._schedule_factory(
+            offset=offset,
+            experience=None,
+        )
+
+        self.task.apply(kwargs=dict(
+            site_id=self.site_config.site.id, target_day_str=serialize(target_day), day_offset=offset,
+            bin_num=self._calculate_bin_for_user(schedule.enrollment.user),
+        ))
+
+        self.assertTrue(mock_ace.send.called)
```

Comment:
Can we refactor the 3 copies of the experience-type tests into a single test base mixin?  The structure of the tests seem to be all the same.  Only the list of test cases passed into `ddt` may differ - which can be declared in the test subclass.

Reply:
Done (kinda)

I couldn't figure out how to get ddt to play nice with the inheritance.